keeper/msg_server_vaults_test.go

```diff
@@ -1890,3 +1891,260 @@ func TestFlexibleVaultRewardsEarlyExit(t *testing.T) {
+		Signer: bob.Address,
+		Vault:  vaults.VaultType(vaultsv1.VaultType_STAKED),
+		Amount: math.NewInt(1 * ONE),
+	})
+	assert.NoError(t, err)
+
+	// ACT: Bob attempts to withdraw 1 USDN from the Flexible Vault.
+	_, err = vaultsServer.Unlock(ctx, &vaults.MsgUnlock{
+		Signer: bob.Address,
+		Vault:  vaults.VaultType(vaultsv1.VaultType_FLEXIBLE),
+		Amount: math.NewInt(1 * ONE),
+	})
+	assert.Error(t, err)
+	assert.Equal(t, "unlock is paused: action is paused", err.Error())
+
+	// ACT: Bob attempts to withdraw 1 USDN from the Staked Vault.
+	_, err = vaultsServer.Unlock(ctx, &vaults.MsgUnlock{
+		Signer: bob.Address,
+		Vault:  vaults.VaultType(vaultsv1.VaultType_STAKED),
+		Amount: math.NewInt(1 * ONE),
+	})
+	assert.NoError(t, err)
+
+	// ARRANGE: Increase the index.
+	err = k.UpdateIndex(ctx, 1.4e12)
+	assert.NoError(t, err)
+}
+
+func TestStakedVaultInterimPeriod(t *testing.T) {
+	account := mocks.AccountKeeper{
+		Accounts: make(map[string]sdk.AccountI),
+	}
+	bank := mocks.BankKeeper{
+		Balances: make(map[string]sdk.Coins),
+	}
+	k, _, ctx := mocks.DollarKeeperWithKeepers(t, bank, account)
+	bank.Restriction = k.SendRestrictionFn
+	k.SetBankKeeper(bank)
+
+	server := keeper.NewMsgServer(k)
+	vaultsServer := keeper.NewVaultsMsgServer(k)
+	vaultsQueryServer := keeper.NewVaultsQueryServer(k)
+	bob := utils.TestAccount()
+	vaultsInterimPeriodYieldCollectorAddress := k.GetVaultsInterimPeriodYieldCollector().String()
+	ctx = ctx.WithHeaderInfo(header.Info{Time: time.Date(2020, 1, 0, 0, 0, 0, 0, time.UTC)})
+	_ = k.UpdateIndex(ctx, 1e12)
+
+	// ARRANGE: Bob mints 100 USDN.
+	_ = k.Mint(ctx, bob.Bytes, math.NewInt(100*ONE), nil)
+
+	// ACT: Bob deposits 50 USDN into the Staked Vault.
+	_, err := vaultsServer.Lock(ctx, &vaults.MsgLock{
+		Signer: bob.Address,
+		Vault:  vaults.VaultType(vaultsv1.VaultType_STAKED),
+		Amount: math.NewInt(50 * ONE),
+	})
+	require.NoError(t, err)
+	require.Equal(t, bank.Balances[bob.Address].AmountOf("uusdn"), math.NewInt(50*ONE)) // 50 USDN.
+
+	// ASSERT: Collector balance is empty.
+	require.Equal(t, bank.Balances[vaultsInterimPeriodYieldCollectorAddress].AmountOf("uusdn"), math.NewInt(0*ONE))
+
+	// ARRANGE: Vaults Program Ends!
+	ctx = ctx.WithHeaderInfo(header.Info{Time: time.Date(2031, 1, 3, 0, 0, 0, 0, time.UTC)})
+
+	k.BeginBlocker(ctx)
+
+	// ASSERT: Matching Vaults Stats state.
+	stats, _ := vaultsQueryServer.Stats(ctx, &vaults.QueryStats{})
+	require.Equal(t, stats.StakedTotalUsers, uint64(1))
+	require.Equal(t, stats.StakedTotalPrincipal, math.NewInt(50*ONE))
+
+	// ARRANGE: Increase the index from 1.0 to 1.1 (~10%).
+	err = k.UpdateIndex(ctx, 1.1e12)
+	require.NoError(t, err)
+
+	// ASSERT: Collector received 5$ from the Staked Vault rewards.
+	require.Equal(t, bank.Balances[vaultsInterimPeriodYieldCollectorAddress].AmountOf("uusdn"), math.NewInt(5*ONE))
+
+	// ACT: Collector claims the yield.
+	_, err = server.ClaimYield(ctx, &types.MsgClaimYield{
+		Signer: bob.Address,
```

Comment:
`Signer` should be the collector address?

Reply:
good catch!